### ISSUE TITLE
Cannot set DateTime

### ISSUE THREAD

**Issue by ludoo** [maxatome/go-vitotrol#9]
Probably a stupid format error on my side, but I can't seem to set DateTime:

```
vitotrol -verbose set DateTime "2019-05-11 07:22:00"
Working with device VT 200 (HO1A / HO1B)@[id]
*** WriteData failed: Unexpected status 9
```

**Comment by maxatome** [comment]
Hi, it is not an error due to the date format, but to the protocol. I cannot test right now, but are you able to get some attributes values? Just to check whether the problem happens only for this attribute or for any.

**Comment by ludoo** [comment]
Hey, thanks for the hyper-quick reply. :)

I'm not sure what you mean with getting attribute values, is it this:

```
vitotrol get DateTime
DateTime: 2019-05-11 07:28:46@2019-05-11 07:28:35 (Current date and time)
```

**Comment by maxatome** [comment]
OK, and do you succeed to set, this time, another attribute, PartyMode or EnergySavingMode for example ?

**Comment by ludoo** [comment]
Yes, that works:

```
vitotrol set PartyMode enabled
vitotrol rget PartyMode
PartyMode: enabled@2019-05-11 09:00:27 (Party mode status)
vitotrol set PartyMode disabled
vitotrol rget PartyMode
PartyMode: disabled@2019-05-11 09:01:03 (Party mode status)
```

**Comment by maxatome** [comment]
And I suppose setting DateTime always fails?
If yes I have to investigate further to unblock you, but I cannot before 2 or 3 days.

**Comment by ludoo** [comment]
Yes it always fails. I tried looking at how the vitodata100 site does it, and while I did it in a hurry and might have been worng, I seem to remember the format used dashes for the time format: HH-MM-SS. Of course, passing in that value to your program results in a format error.

I am in absolutely no hurry, your program is fantastic and much more useful than the stock ones, and this is a tiny piece of functionality (although useful as my system's time tends to drift and resets itself quite a bit).

**Comment by ludoo** [comment]
PS - I don't know Go well enough to try looking at this myself, unfortunately.

**Comment by maxatome** [comment]
Could you try to change all ":" to "-" at https://github.com/maxatome/go-vitotrol/blob/master/time.go#L8 and tell me if it works?

**Comment by ludoo** [comment]
```./vitotrol set DateTime "2019-05-11 20-08-00"
*** WriteData error: Unbekannter Fehler
String was not recognized as a valid DateTime. [#99999]
```

**Comment by maxatome** [comment]
You can restore the datetime format as it seems ":" are correct.
At https://github.com/maxatome/go-vitotrol/blob/master/device.go#L558 you can replace `status != 4` by `false` and check date/time is well updated, but I am afraid that 9 status could be an error code telling us something like "unknown error" or "this cannot be achieved"... To test...

**Comment by ludoo** [comment]
Hmmmm that seems to work.

I changed that line to be

`if status != 4 && status != 9 {`

and this is why it seems to work:

```
./vitotrol rget DateTime
DateTime: 2019-05-12 08:34:05@2019-05-12 08:32:13 (Current date and time)

./vitotrol set DateTime "2019-05-12 08:30:00"

./vitotrol rget DateTime
DateTime: 2019-05-12 08:30:19@2019-05-12 08:32:59 (Current date and time)
```

**Comment by maxatome** [comment]
Interesting. Could you file a PR referencing this issue, with this change + modifying the comment just above? Thanks!

**Comment by maxatome** [comment]
Fixed by #10, thanks again. Would be cool to have the exact meaning for each status code, but I failed to found any doc referencing them.
